### PR TITLE
Remove pending version state.

### LINKED ISSUE #806: Remove Pending State for Version, Category, and Entry
It turns out the approach to have pending state doesn't work properly. After discussion, @timlinux suggests removing the pending state for:
- [x] Version
- [x] Category
- [x] Entry

It will have these consequents:
- [x]  All normal user can not edit or add anything
- [x] Edit will happen directly after click save (no more pending state)
- [x] Only staff, superuser, project owner and version manager can make edits

### PR DESCRIPTION
**Pull request by ismailsunni**
Fix #806

### CODE REVIEW

File: django_project/changes/tests/test_models.py [hunk 1]
```diff
@@ -193,8 +192,8 @@ def test_formatted_release_date(self):
         """Tests we can get, set and present the release date nicely."""
         model = VersionF.create(
             description=u'New description',
-            release_date=datetime(2016, 06, 06),
-            approved=True)
+            release_date=datetime(2016, 06, 06)
```
Comment (ismailsunni) [inline django_project/changes/tests/test_models.py]: __[Flake8]__

[E999] SyntaxError: invalid token

**Comment by ann26** [comment]
LGTM, but the codecov is decreasing

**Comment by ismailsunni** [comment]
> LGTM, but the codecov is decreasing

The codecov is decreasing since there was not unit test previously on the file that I changed.

**Comment by timlinux** [comment]
LGTM

File: django_project/changes/feeds/version.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def items(self, obj):
         :param obj: A project
         :type obj: Project
 
-        :returns: List of approved version of a project
+        :returns: List of all version of a project
         :rtype: list
         """
         return Version.objects.filter(project=obj, approved=True).order_by(
```
Comment (Gustry) [inline django_project/changes/feeds/version.py]: approved?

Comment (ismailsunni) [reply django_project/changes/feeds/version.py]: Thanks, @Gustry !!

**Pull request merged**